### PR TITLE
Fix/2447 dynamic datatable fullscreen height

### PR DESCRIPTION
**Pull request by kwongz**
### Checklist

- ~~For UI or styling changes, I have added a screenshot or gif showing before & after~~
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- ~~I have added to the docs where applicable~~
- ~~I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable~~

### CODE REVIEW

**Comment by kwongz** [comment]
fix #2447

File: packages/ui/core-components/src/lib/unsorted/viz/table/_DataTable.svelte [hunk 1]
```diff
@@ -437,7 +437,8 @@
 	<Fullscreen bind:open={fullscreen} {search}>
 		<!-- header and last row are 22.5+22.5 = 45px, middle rows are 23 -->
 		{@const ROW_HEIGHT = 23}
```
Comment (archiewood) [inline packages/ui/core-components/src/lib/unsorted/viz/table/_DataTable.svelte:439]: will this work if someone uses `<DataTable compact=true/>`

Comment (kwongz) [reply packages/ui/core-components/src/lib/unsorted/viz/table/_DataTable.svelte:439]: No issues in Chrome and Firefox, User is experiencing issue in Chrome

Chrome:
image

image

**Comment by kwongz** [comment]
@archiewood Have not been able to reproduce to the degree the user was experiencing, but i've left a note in slack to get their markdown code of the `DataTable` and asked about any custom CSS styling they may have used.

Currently, on our Docs site, when the search bar is present, we do experience a bit of overflow (similar, but not to the debilitating degree our user has flagged)

(when search bar is present, slight overflow is preset indicated by y scroll bar)
image

This PR aims to take the search bar's additional height into its total `DataTable`'s height, which will remove the slight overflow we are experiencing.

After speaking with the user, returning the overflow-y property to the Fullscreen DataTable will enable the user to access the fullscreen datable pagination. The User confirmed that this was what they desired to return.

**Comment by kwongz** [comment]
Added Comments/Variables to clarify calculations in height calc

**Review by mcrascal — APPROVED** [review]

**Pull request merged**